From 6eabb24063a70305d73566e27ec52769e3bf0cf0 Mon Sep 17 00:00:00 2001
From: Allan Stephens <allan.stephens@windriver.com>
Date: Thu, 6 Oct 2011 13:57:51 -0400
Subject: [PATCH 052/183] tipc: Enable use by containers having their own
 network namespace
Git-commit: 6eabb24063a70305d73566e27ec52769e3bf0cf0
Patch-mainline: Merged into tipc-devel
References: bnc#797455

Permits a Linux container to use TIPC sockets even when it has its own
network namespace defined by removing the check that prohibits such use.
This makes it possible for users who wish to isolate their container
network traffic from normal network traffic to utilize TIPC.

Signed-off-by: Allan Stephens <allan.stephens@windriver.com>
Signed-off-by: Paul Gortmaker <paul.gortmaker@windriver.com>
Signed-off-by: Erik Hugne <erik.hugne@ericsson.com>
Signed-off-by: Jiri Kosina <jkosina@suse.cz>

---
 net/tipc/socket.c |    3 ---
 1 files changed, 0 insertions(+), 3 deletions(-)

diff --git a/net/tipc/socket.c b/net/tipc/socket.c
index 9440a3d..7ed03d8 100644
--- a/net/tipc/socket.c
+++ b/net/tipc/socket.c
@@ -184,9 +184,6 @@ static int tipc_create(struct net *net, struct socket *sock, int protocol,
 
 	/* Validate arguments */
 
-	if (!net_eq(net, &init_net))
-		return -EAFNOSUPPORT;
-
 	if (unlikely(protocol != 0))
 		return -EPROTONOSUPPORT;
 
-- 
1.7.8.3

